### ISSUE TITLE
zmq socket closed before sending all data

### ISSUE THREAD

**Issue by tsaridas** [saltstack/salt#50393]
### Description of Issue/Question
While testing sending command to multiple minions (200+) at the same time we noticed that some syndic randomly did not send publish the event to the minions and tracked down the issue to the ZeroMQPubServerChannel class.

After adding debug log after

https://github.com/saltstack/salt/blob/develop/salt/transport/zeromq.py#L938

the problem seemed to be fixed thinking that the socket is closed before all events are send.

Worth noticing that we have zmq_filtering enabled on the master and minion side so an event is created for each minion.

We switched from zmq socket to normal sockets ( AF_UNIX, SOCK_DGRAM )  for publishing events and the issue seems to have been resolved.

https://github.com/saltstack/salt/blob/develop/salt/transport/zeromq.py#L909
https://github.com/saltstack/salt/blob/develop/salt/transport/zeromq.py#L910

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
2016.11.10

**Comment by garethgreenaway** [comment]
@saltstack/team-core Anyone have thoughts on this one?

**Comment by cachedout** [comment]
@garethgreenaway Right now this is something that @dwoz is looking into.

**Comment by garethgreenaway** [comment]
@cachedout Thanks.  Assigned to @dwoz.

**Comment by dwoz** [comment]
@tsaridas can you please provide us the output of `salt --versions-report` ?

**Comment by tsaridas** [comment]
```
Salt Version:

           Salt: 2016.11.10


Dependency Versions:

           cffi: Not Installed

       cherrypy: 3.2.2

       dateutil: 1.4.1

      docker-py: Not Installed

          gitdb: Not Installed

      gitpython: Not Installed

          ioflo: Not Installed

         Jinja2: 2.7.3

        libgit2: Not Installed

        libnacl: Not Installed

       M2Crypto: 0.20.2

           Mako: Not Installed

   msgpack-pure: Not Installed

msgpack-python: 0.4.8

   mysql-python: Not Installed

      pycparser: Not Installed

       pycrypto: 2.6.1

   pycryptodome: 3.4.3

         pygit2: Not Installed

         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)

   python-gnupg: Not Installed

         PyYAML: 3.11

          PyZMQ: 14.5.0

           RAET: Not Installed

          smmap: Not Installed

        timelib: Not Installed

        Tornado: 4.2.1

            ZMQ: 4.0.5


System Versions:

        dist: redhat 6.8 Santiago

        machine: x86_64

        release: 2.6.

       system: Linux

        version: Red Hat Enterprise Linux Server 6.8 Santiago
```

**Comment by dwoz** [comment]
@tsaridas I've submitted a PR with my proposed fix #50424. If you would like to test it and provide a review that would be helpful.
